From 13679185d7e6393fb7cfaac98c58a415aec651a1 Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Fri, 25 Oct 2019 12:54:48 +0300
Subject: [PATCH 21/26] kernel/dm: Include <sys/param.h> first.

 Allow the <dev/disk/dm/dm.h> to include common headers first.
---
 sys/dev/disk/dm/dm_ioctl.c                  | 1 +
 sys/dev/disk/dm/dm_table.c                  | 1 +
 sys/dev/disk/dm/dm_target.c                 | 1 +
 sys/dev/disk/dm/linear/dm_target_linear.c   | 4 ++--
 sys/dev/disk/dm/striped/dm_target_striped.c | 4 ++--
 5 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/sys/dev/disk/dm/dm_ioctl.c b/sys/dev/disk/dm/dm_ioctl.c
index 48373a709a..1821ca1490 100644
--- a/sys/dev/disk/dm/dm_ioctl.c
+++ b/sys/dev/disk/dm/dm_ioctl.c
@@ -79,6 +79,7 @@
  *
  */
 
+#include <sys/param.h>
 #include <sys/device.h>
 #include <sys/malloc.h>
 #include <cpu/atomic.h>
diff --git a/sys/dev/disk/dm/dm_table.c b/sys/dev/disk/dm/dm_table.c
index a10b4d3173..2c69c42b94 100644
--- a/sys/dev/disk/dm/dm_table.c
+++ b/sys/dev/disk/dm/dm_table.c
@@ -30,6 +30,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <sys/param.h>
 #include <sys/malloc.h>
 #include <cpu/atomic.h>
 #include <dev/disk/dm/dm.h>
diff --git a/sys/dev/disk/dm/dm_target.c b/sys/dev/disk/dm/dm_target.c
index 4bcc732270..5402907eb1 100644
--- a/sys/dev/disk/dm/dm_target.c
+++ b/sys/dev/disk/dm/dm_target.c
@@ -30,6 +30,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <sys/param.h>
 #include <sys/malloc.h>
 #include <sys/module.h>
 #include <sys/linker.h>
diff --git a/sys/dev/disk/dm/linear/dm_target_linear.c b/sys/dev/disk/dm/linear/dm_target_linear.c
index 745a0e1072..104c2771e0 100644
--- a/sys/dev/disk/dm/linear/dm_target_linear.c
+++ b/sys/dev/disk/dm/linear/dm_target_linear.c
@@ -34,9 +34,9 @@
  * This file implements initial version of device-mapper dklinear target.
  */
 
-#include <sys/malloc.h>
-
 #include <dev/disk/dm/dm.h>
+#include <sys/malloc.h>		/* for malloc macros, dm.h includes sys/param.h */
+
 MALLOC_DEFINE(M_DMLINEAR, "dm_linear", "Device Mapper Target Linear");
 
 typedef struct target_linear_config {
diff --git a/sys/dev/disk/dm/striped/dm_target_striped.c b/sys/dev/disk/dm/striped/dm_target_striped.c
index 241c250523..c69cb677de 100644
--- a/sys/dev/disk/dm/striped/dm_target_striped.c
+++ b/sys/dev/disk/dm/striped/dm_target_striped.c
@@ -37,9 +37,9 @@
  *
  * DragonFly changes: Increase to an unlimited number of stripes
  */
-#include <sys/malloc.h>
-
 #include <dev/disk/dm/dm.h>
+#include <sys/malloc.h>		/* for malloc macros, dm.h includes sys/param.h */
+
 MALLOC_DEFINE(M_DMSTRIPE, "dm_striped", "Device Mapper Target Striped");
 
 #define MAX_STRIPES 32
-- 
2.23.0

